RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  BaseManageSubscriptionViewModel.swift
+//
+//  Created by Facundo Menzella on 5/5/25.
+
+import Foundation
+@_spi(Internal) import RevenueCat
+import SwiftUI
+
+#if os(iOS)
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@MainActor
+class BaseManageSubscriptionViewModel: ObservableObject {
+
+    let screen: CustomerCenterConfigData.Screen
+
+    var relevantPathsForPurchase: [CustomerCenterConfigData.HelpPath] {
+        paths.relevantPathsForPurchase(purchaseInformation)
+    }
+
+    @Published
+    var showAllPurchases = false
+
+    @Published
+    var showRestoreAlert: Bool = false
+
+    @Published
+    var restoreAlertType: RestorePurchasesAlertViewModel.AlertType
+
+    @Published
+    var feedbackSurveyData: FeedbackSurveyData?
+
+    @Published
+    var loadingPath: CustomerCenterConfigData.HelpPath?
+
+    @Published
+    var promotionalOfferData: PromotionalOfferData?
+
+    @Published
+    var inAppBrowserURL: IdentifiableURL?
+
+    let actionWrapper: CustomerCenterActionWrapper
+
+    @Published
+    var purchaseInformation: PurchaseInformation?
+
+    @Published
+    private(set) var refundRequestStatus: RefundRequestStatus?
+
+    private var error: Error?
+    private let loadPromotionalOfferUseCase: LoadPromotionalOfferUseCaseType
+    private let paths: [CustomerCenterConfigData.HelpPath]
+    private(set) var purchasesProvider: CustomerCenterPurchasesType
+
+    init(
+        screen: CustomerCenterConfigData.Screen,
+        actionWrapper: CustomerCenterActionWrapper,
+        purchaseInformation: PurchaseInformation? = nil,
+        refundRequestStatus: RefundRequestStatus? = nil,
+        purchasesProvider: CustomerCenterPurchasesType,
+        loadPromotionalOfferUseCase: LoadPromotionalOfferUseCaseType? = nil) {
+            self.screen = screen
+            self.paths = screen.filteredPaths
+            self.purchaseInformation = purchaseInformation
+            self.purchasesProvider = purchasesProvider
+            self.refundRequestStatus = refundRequestStatus
+            self.actionWrapper = actionWrapper
+            self.loadPromotionalOfferUseCase = loadPromotionalOfferUseCase
+            ?? LoadPromotionalOfferUseCase(purchasesProvider: purchasesProvider)
+            self.restoreAlertType = .loading
+        }
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    func determineFlow(for path: CustomerCenterConfigData.HelpPath, activeProductId: String? = nil) async {
```

Comment:
nitpick, but really is a handler, not a decision maker, right? based on the `determineFlow` name I expected a non-nil return type, then I realized that the method itself already handles the logic. Maybe we can reflect that with the name?

Reply:
Wen with `handleHelpPath:for:`